runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/EngineBoss.java

```diff
@@ -0,0 +1,181 @@
+        EngineConfiguration config,
+        ErrorHandler errorHandler,
+        Collection<Binding> bindings)
+    {
+        this.poller = new Poller();
+        this.taskQueue = new ConcurrentLinkedDeque<>();
+
+        this.controllersByType = bindings.stream()
+        .flatMap(b ->
+        {
+            BindingController controller = b.supply(this);
+            return controller != null ? Stream.of(Map.entry(b.name(), controller)) : Stream.empty();
+        })
+        .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        IdleStrategy idleStrategy = new BackoffIdleStrategy(
+            config.maxSpins(),
+            config.maxYields(),
+            config.minParkNanos(),
+            config.maxParkNanos());
+
+        this.runner = new AgentRunner(idleStrategy, errorHandler, null, this);
+    }
+
+    @Override
+    public int doWork() throws Exception
+    {
+        int workDone = 0;
+
+        try
+        {
+            workDone += poller.doWork();
+
+            if (!taskQueue.isEmpty())
+            {
+                for (Runnable task = taskQueue.poll(); task != null; task = taskQueue.poll())
+                {
+                    task.run();
+                    workDone++;
+                }
+            }
+        }
+        catch (Throwable ex)
+        {
+            throw new AgentTerminationException(ex);
+        }
+
+        return workDone;
+    }
+
+    public void doStart()
+    {
+        thread = startOnThread(runner, Thread::new);
+    }
+
+    public void doClose()
+    {
+        try
+        {
+            controllersByType.values().forEach(BindingController::detachAll);
+            controllersByType.clear();
+
+            poller.close();
+
+            Consumer<Thread> timeout = t -> rethrowUnchecked(new IllegalStateException("close timeout"));
+            runner.close((int) SECONDS.toMillis(5L), timeout);
+        }
+        finally
+        {
+            thread = null;
+        }
+    }
+
+    public CompletableFuture<Void> attach(
+        NamespaceConfig namespace)
+    {
+        NamespaceTask attachedTask = new NamespaceTask(namespace, this::attachNamespace);
+        taskQueue.offer(attachedTask);
+
+        return attachedTask.future();
+    }
+
+    private void attachNamespace(
+        NamespaceConfig namespace)
+    {
+        for (BindingConfig binding : namespace.bindings)
+        {
+            BindingController controller = controllersByType.get(binding.type);
+            if (controller != null && binding.kind == KindConfig.SERVER)
+            {
+                controller.attach(binding);
+            }
+        }
+    }
+
+    public void detach(
+        NamespaceConfig namespace)
+    {
+        for (BindingConfig binding : namespace.bindings)
+        {
+            BindingController controller = controllersByType.get(binding.type);
+            if (controller != null)
+            {
+                controller.detach(binding);
+            }
+        }
```

Comment:
I think we need to offer this as a `NamespaceTask` too, to make sure it executes on the boss thread, not the caller thread.